api/types/registry/registry.go

```diff
@@ -17,23 +17,6 @@ type ServiceConfig struct {
 	ExtraFields map[string]any `json:"-"`
 }
 
-// MarshalJSON implements a custom marshaler to include legacy fields
-// in API responses.
-func (sc *ServiceConfig) MarshalJSON() ([]byte, error) {
```

Comment:
It's used by the daemon to produce API responses for older API versions. If we remove this here, we probably need to create a wrapper struct with marshaling in the daemon API server; see https://github.com/moby/moby/blob/ed8a6a89cad73af1e54eb4e94bdf7c0f46a5737b/daemon/server/router/system/system_routes.go#L105-L112

- https://github.com/moby/moby/pull/49749

Reply:
I recall I had a branch somewhere with a more generic package to allow "append old API fields" but not sure if I have preserved it 🙈